src/components/vr_cooperation/src/command/base_gpb_request.cc

```diff
@@ -75,12 +75,18 @@ void BaseGpbRequest::SendMessageToHMI(
   EventDispatcher<vr_hmi_api::ServiceMessage, vr_hmi_api::RPCName>::instance()
       ->add_observer(message.rpc(), message.correlation_id(), this);
   parent_->SendMessageToHMI(message);
+  if (vr_hmi_api::RESPONSE == message.rpc_type()) {
+    parent_->UnregisterRequest(message.correlation_id());
```

Comment:
In previous commit I already remove it from VR Module's functions and mention the reason also. And I personal think the logic of delete request should be in BasexxxRequest class.